spacy/gold/corpus.py

```diff
@@ -151,3 +159,53 @@ def read_docbin(
                         i += 1
                         if self.limit >= 1 and i >= self.limit:
                             break
+
+
+class JsonlTexts:
+    """Iterate Doc objects from a file or directory of jsonl 
+    formatted raw text files.
+
+    path (Path): The directory or filename to read from.
+    min_length (int): Minimum document length (in tokens). Shorter documents
+        will be skipped. Defaults to 0, which indicates no limit.
+ 
+    max_length (int): Maximum document length (in tokens). Longer documents will
+        be skipped. Defaults to 0, which indicates no limit.
+    limit (int): Limit corpus to a subset of examples, e.g. for debugging.
+        Defaults to 0, which indicates no limit.
+
+    DOCS: https://nightly.spacy.io/api/corpus
+    """
+    file_type = "jsonl"
+
+    def __init__(
+        self,
+        path: Union[str, Path],
+        *,
+        limit: int = 0,
+        min_length: int = 0,
+        max_length: int = 0,
+    ) -> None:
+        self.path = util.ensure_path(path)
+        self.min_length = min_length
+        self.max_length = max_length
+        self.limit = limit
+
+    def __call__(self, nlp: "Language") -> Iterator[Doc]:
```

Comment:
Can't this return `Iterator[Example]` instead? So we don't need the `ensure_docs` stuff? And then the `registry.readers` would contain readers with the same return type... (I guess I'm scared this type of "doc or example" will trickle back into the main codebase as it did before)

Reply:
This can, yeah. I do like the idea of letting users iterate out `Doc` objects, constructing an `Example` seems sort of bureaucratic? I don't really mind though.

For now I'll just change this to produce `Example`, and I'll leave the `ensure_docs` in.